commands/repl.js

```diff
@@ -1,20 +1,86 @@
-           '    near repl --accountId bob\n    > console.log(account)\n    > .load script.js',
-    builder: (yargs) => yargs,
+           '    near repl --accountId bob\n    > console.log(account)\n    > .load script.js\n\n' +
+           '    near repl -s ./scripts/delpoyContracts.js\n' +
+           '    near repl -s ./scripts/run.js -- test.near bob.test.near\n' + 
+           '    near repl -s ./script.ts \n',
+    builder: (yargs) => yargs
+        .option('script', {
+            desc: '\'repl\': Path to a JS/TS file which exports an async function `main`,\nwhich takes a `context` object.',
+            type: 'string',
+            alias: 's',
+        }),
     handler: async (argv) => {
+        const extraArgs = argv._.slice(1);
+        const { script, accountId } = argv;
         await eventtracking.askForConsentIfNeeded(argv);
-        const repl = require('repl');
-        const context = repl.start('> ').context;
-        context.nearAPI = require('near-api-js');
-        context.near = await require('../utils/connect')(argv);
-        if (argv.accountId) {
-            context.account = await context.near.account(argv.accountId);
+        const nearContext = {
+            nearAPI: require("near-api-js"),
+            near: await require("../utils/connect")(argv),
+        };
+        if (accountId) {
+            nearContext.account = await nearContext.near.account(accountId);
+        }
+        if (extraArgs.length > 0) {
+            nearContext.argv = extraArgs;
+        }
+        if (script) {
+            const scriptModule = loadScript(script);
+            try {
+                await scriptModule.main(nearContext);
+            } catch (error) {
+                console.error(`${script} Failed\n`);
+                console.error(error);
+                process.exit(1);
+            }
+        } else {
+            const repl = require("repl");
+            const replContext = repl.start("> ").context;
+            const { near, nearAPI, account } = nearContext;
+            replContext.nearAPI = nearAPI;
+            replContext.near = near;
+            if (account) {
+                replContext.account = account;
+            }
         }
     }
 };
+
+function loadScript(script) {
+    const scriptPath = script.startsWith(".")
+        ? resolve(process.cwd(), script)
+        : script;
+    try {
+        if (scriptPath.endsWith(".ts")) {
+            loadTs(scriptPath);
+        }
+        return require(scriptPath);
+    } catch (error) {
+        console.error(`Failed to load ${scriptPath}.\n`);
+        console.error(error);
+        process.exit(1);
+    }
+}
+
+function loadTs(scriptPath) {
+    try {
+        require("ts-node").register({ transpileOnly: true });
+    } catch {
+        console.error(
+            `Failed to load \`ts-node\` for typescript file ${scriptPath}. Probably need to install \`ts-node\` and \`typescript\`.\n`
+        );
+        console.error(error);
+        process.exit(1);
```

Comment:
```suggestion
        throw error;
```

Reply:
I found that throwing the error would cause all documentation to print.